aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1315,7 +1327,7 @@ impl AptosVM {
 
     pub(crate) fn process_waypoint_change_set(
         &self,
-        resolver: &impl MoveResolverExt,
+        resolver: &impl AptosMoveResolver,
```

Comment:
My intention was to hide view accesses from `AptosMoveResolver`, e.g. you should not be able to call `get_state_value` or similar because that wold bypass adapter's configs and seems error-prone. But the boundary here is quite thin, I am wondering if we can completely merge the two in the future